src/Bridges/Bridges.jl

```diff
@@ -23,17 +23,70 @@ include("debug.jl")
 """
     full_bridge_optimizer(model::MOI.ModelLike, ::Type{T}) where {T}
 
-Returns a `LazyBridgeOptimizer` bridging `model` for every bridge defined in
-this package and for the coefficient type `T`.
+Returns a [`LazyBridgeOptimizer`](@ref) bridging `model` for every bridge
+defined in this package (see below for the few exceptions) and for the
+coefficient type `T` in addition to the bridges in the list returned by
+`MOI.get(model, MOI.Bridges.ToAdd{T}())`.
+
+See also [`ToAdd`](@ref).
+
+!!! note
+    The following bridges are not added by `full_bridge_optimizer` except if
+    they are in the list returned by `MOI.get(model, MOI.Bridges.ToAdd{T}())`
+    (see the docstrings of the corresponding bridge for the reason they are not
+    added):
+    * [`Constraint.SOCtoNonConvexQuad`](@ref),
+      [`Constraint.RSOCtoNonConvexQuad`](@ref) and
+      [`Constraint.SOCtoPSDBridge`](@ref).
+    * The subtypes of [`Constraint.AbstractToIntervalBridge`](@ref) (i.e.
+      [`Constraint.GreaterToIntervalBridge`](@ref) and
+      [`Constraint.LessToIntervalBridge`](@ref)) if `T` is not a subtype of
+      `AbstractFloat`.
 """
 function full_bridge_optimizer(model::MOI.ModelLike, ::Type{T}) where {T}
     bridged_model = LazyBridgeOptimizer(model)
+    for BT in MOI.get(model, ToAdd{T}())
+        MOIB.add_bridge(bridged_model, BT)
+    end
     Variable.add_all_bridges(bridged_model, T)
     Constraint.add_all_bridges(bridged_model, T)
     Objective.add_all_bridges(bridged_model, T)
     return bridged_model
 end
 
+"""
+    ToAdd{T}()
+
+A list of additional bridges that should be added to the
+[`LazyBridgeOptimizer`](@ref) in `full_bridge_optimizer(optimizer, T)`.
+
+See also [`full_bridge_optimizer`](@ref).
+
+## Examples
+
+Suppose an optimizer can exploit specific structure of a constraint.
+For instance, suppose it can exploit the structure of the matrix `A`
+in the linear system of equations `A * x = b`.
+The optimizer can create a function:
+```julia
+struct MatrixAffineFunction{T,M<:AbstractMatrix{T}} <: MOI.AbstractVectorFunction
+    A::M
+    b::Vector{T}
+end
+```
+and then a bridge from `VectorAffineFunction{T}` to
+`MatrixAffineFunction{T,SparseMatrixCSC{T,Int}}`.
+The bridge is then included in the list returned as the value of the
+`ToAdd` attribute. That way, the optimizer can exploit a specific matrix
+structure if the user uses a `MatrixAffineFunction` and in case the user uses
+`VectorAffineFunction` then it's automatically bridged.
+"""
+struct ToAdd{T} <: MOI.AbstractOptimizerAttribute end
```

Comment:
I'm not a fan of `ToAdd`. How about

````Julia
"""
    ListOfNonstandardBridges{T}() <: MOI.AbstractOptimizerAttribute

Any optimizer can be wrapped in a [`LazyBridgeOptimizer`](@ref) using 
[`full_bridge_optimizer`](@ref). However, by default [`LazyBridgeOptimizer`](@ref)
uses a limited set of bridges that are:

  1. implemented in `MOI.Bridges`
  2. generally applicable for all optimizers.

For some optimizers however, it is useful to add additional bridges, such as
those that are implemented in external packages (e.g., within the solver package
itself) or only apply in certain circumstances (e.g., 
[`Constraint.SOCtoNonConvexQuad`](@ref)).

Optimizers should implement the `ListOfNonstandardBridges` attribute to return a
vector of bridge types that are added by [`full_bridge_optimizer`](@ref) in
addition to the list of default bridges.

Note that optimizers implementing `ListOfNonstandardBridges` may be unusable
ifthe non-standard bridges are not added.

## Examples

### An optimizer using a non-default bridge in `MOI.Bridges`

Solvers supporting [`MOI.ScalarQuadraticFunction`](@ref) can support
[`MOI.SecondOrderCone`](@ref) and [`MOI.RotatedSecondOrderCone`](@ref) by
defining:
```julia
function MOI.get(::MyQuadraticOptimizer, ::ListOfNonstandardBridges{Float64})
    return Type[
        MOI.Bridges.Constraint.SOCtoNonConvexQuad{Float64},
        MOI.Bridges.Constraint.RSOCtoNonConvexQuad{Float64},
    ]
end
```

### An optimizer defining an internal bridge

Suppose an optimizer can exploit specific structure of a constraint, e.g., it
can exploit the structure of the matrix `A` in the linear system of equations
`A * x = b`.

The optimizer can define the function:
```julia
struct MatrixAffineFunction{T} <: MOI.AbstractVectorFunction
    A::SomeStructuredMatrixType{T}
    b::Vector{T}
end
```
and then a bridge
```julia
struct MatrixAffineFunctionBridge{T} <: MOI.Constraint.AbstractBridge
    f::MOI.VectorAffineFunction{T}
end
```
from `VectorAffineFunction{T}` to the `MatrixAffineFunction`. Finally, it
defines:
```julia
function MOI.get(::Optimizer{T}, ::ListOfNonstandardBridges{T}) where {T}
    return Type[MatrixAffineFunctionBridge{T}]
end
```
"""
````